rules/rules.js

```diff
@@ -207,11 +207,19 @@ let JSRule = function (ruleConfig) {
     let SimpleRule = Java.extend(Java.type('org.openhab.core.automation.module.script.rulesupport.shared.simple.SimpleRule'));
 
     let doExecute = function (module, input) {
-        try {
-            return ruleConfig.execute(getTriggeredData(input));
-        } catch (error) {
-            throw Error(`Failed to execute rule ${ruid}: ${error}: ${error.stack}`);
+      try {
+        return ruleConfig.execute(getTriggeredData(input));
+      } catch (error) {
+        // logging error is required for meaningful error log message
+        // when throwing error: error is caught by core framework and no meaningful message is logged
+        if (error.stack) {
+          console.error(`Failed to execute rule ${ruid}: ${error}: ${error.stack}`);
+          throw Error(`Failed to execute rule ${ruid}: ${error}: ${error.stack}`);
+        } else {
+          console.error(`Failed to execute rule ${ruid}: ${error}`);
+          throw Error(`Failed to execute rule ${ruid}: ${error}`);
```

Comment:
Would it be cleaner to:

```
let msg = `Failed to execute rule ${ruid}: ${error} ${error.stack ? ": " + error.stack : "" }`
console.error(msg);
throw Error(msg);
```

So that you don't need to repeat the message so many times?